src/core/execute.c

```diff
@@ -3277,6 +3286,28 @@ static int exec_child(
                         log_unit_warning(unit, "PrivateNetwork=yes is configured, but the kernel does not support network namespaces, ignoring.");
         }
 
+#if HAVE_APPARMOR
+        if (needs_sandboxing) {
+                if (use_apparmor && context->apparmor_profile) {
+                        r = aa_change_onexec(context->apparmor_profile);
+                        if (r < 0 && !context->apparmor_profile_ignore) {
+                                *exit_status = EXIT_APPARMOR_PROFILE;
+                                return log_unit_error_errno(unit, errno, "Failed to prepare AppArmor profile change to %s: %m", context->apparmor_profile);
+                        }
+                }
+        }
+#endif
```

Comment:
i figure moving this one earlier should be OK, since it has no immediate effect afaiu, only on the next execve()... bit then again, my clue about aa is limited, I don#t run this, so i kinda hope there's somebody who tested this.